tests/test_config_section.py

```diff
@@ -0,0 +1,546 @@
+        """
+        self.assertRaises(ValueError, str_to_bool, self.global_config['wrong_bool_value'])
+        result = self.global_config._validate_str_to_bool('wrong_bool_value')
+        self.assertIsNotNone(result)
+        self.assertIsInstance(result, tuple)
+
+    def test_validate_non_existing_key(self):
+        """
+        Test validation of non-existing key 
+        """
+        result = self.global_config._validate_str_to_bool('does_not_exist')
+        self.assertIsNotNone(result)
+        self.assertIsInstance(result, tuple)
+
+    def test_validate_string(self):
+        """
+        Test validation of correct string value 
+        """
+        result = self.global_config._validate_non_empty_string('log_file')
+        self.assertIsNone(result)
+
+    def test_validate_interval(self):
+        """
+        Test validation of time interval
+        """
+        result = self.global_config._validate_interval()
+        self.assertIsNone(result)
+
+    def test_validate_wrong_interval(self):
+        """
+        Test validation of wrong time interval
+        """
+        self.global_config['interval'] = '10'
+        result = self.global_config._validate_interval()
+        self.assertIsNotNone(result)
+        interval = self.global_config.get('interval')
+        self.assertEqual(interval, MinimumSendInterval)
+
+    def test_validate_missing_interval(self):
+        """
+        Test validation of wrong time interval
+        """
+        self.global_config = GlobalSection('global', None)
+        for key, value in GLOBAL_SECTION_VALUES.items():
+            if key != 'interval':
+                self.global_config[key] = value
+        result = self.global_config._validate_interval()
+        self.assertIsNone(result)
+        interval = self.global_config['interval']
+        self.assertEqual(interval, DefaultInterval)
+
+    def test_validate_configs(self):
+        """
+        Test validation of configs (list of paths to config files)
+        """
+        result = self.global_config._validate_configs()
+        self.assertIsNone(result)
+
+    def test_validate_section_values(self):
+        """
+        Test validation of all config values 
+        """
+        validate_messages = self.global_config.validate()
+        print(validate_messages)
+        expected_results = [
+            ('warning', 'Ignoring unknown configuration option "wrong_bool_value"'),
+            ('warning', 'log_per_config must be a valid boolean, using default. See man virt-who-config for more info'),
+            ('warning', 'Value for reporter_id not set, using default'),
+            ('warning', 'Value for log_dir not set, using default'),
+            ('warning', 'background must be a valid boolean, using default. See man virt-who-config for more info')
+        ]
+        self.assertGreater(len(validate_messages), 0)
+        # self.assertEqual(expected_results, validate_messages)
```

Comment:
The format of message is still subject of changes. I will add TODO comment there.